### PR TITLE
Apply react.gradle from node_modules/react-native

### PR DESCRIPTION
**Pull request by satya164**
The goal is to minimize the number of files we need to bootstrap. This allows us to make the upgrade process smoother for everyone.

If someone needs to customize the file, we already provide some config options. The ability to copy the file and modify it is always there for those few who need it.

**Test plan**

Generate a new project with the updated template.  The app should build and run fine both in debug and production mode.

Related #6292

### CODE REVIEW

**Comment by satya164** [comment]
cc @foghina @bestander

File: react.gradle [hunk 1]
```diff
@@ -72,10 +72,10 @@ gradle.projectsEvaluated {
                 // Set up dev mode
                 def devEnabled = !targetName.toLowerCase().contains("release")
                 if (Os.isFamily(Os.FAMILY_WINDOWS)) {
-                    commandLine "cmd", "/c", "node", "node_modules/react-native/local-cli/cli.js", "bundle", "--platform", "android", "--dev", "${devEnabled}",
+                    commandLine "cmd", "/c", "react-native", "bundle", "--platform", "android", "--dev", "${devEnabled}",
```
Comment (foghina) [inline react.gradle]: These changes look unrelated.

File: react.gradle [hunk 2]
```diff
@@ -72,10 +72,10 @@ gradle.projectsEvaluated {
...
                             "--entry-file", entryFile, "--bundle-output", jsBundleFile, "--assets-dest", resourcesDir
                 } else {
-                    commandLine "node", "node_modules/react-native/local-cli/cli.js", "bundle", "--platform", "android", "--dev", "${devEnabled}",
+                    commandLine "react-native", "bundle", "--platform", "android", "--dev", "${devEnabled}",
```
Comment (foghina) [inline react.gradle]: Same as above.

**Comment by foghina** [comment]
Also, perhaps the top level is not the best place `react.gradle`? I think it would be better in `ReactAndroid/`, but if that interferes with the build in any way I'm fine with top level.

File: react.gradle (hunk 2 above)
Comment (satya164) [inline react.gradle]: @foghina Yeah, fixed, will send this in another PR.

**Comment by satya164** [comment]
@foghina We can move it to `ReactAndroid`, but IMHO we shouldn't be shipping `ReactAndroid` to NPM. Adds 5 MB of extra download for no good reason.

**Comment by foghina** [comment]
> we shouldn't be shipping ReactAndroid to NPM

Argh, forgot about that, good point. Looks good, thanks!

@facebook-github-bot shipit

**Comment by bestander** [comment]
Looks like tests got broken, @satya164

**Comment by satya164** [comment]
@bestander Checking

**Comment by satya164** [comment]
@bestander Seem to be unrelated to this change? They are regarding adb issues, like unable to connect device.

**Comment by bestander** [comment]
There are 2 issues:
1. npm bundle test
2. adb gradle install

It is odd that both failed consistently 2 times https://circleci.com/gh/facebook/react-native/tree/pull%2F6610
I'll try rebuilding without caches

**Comment by satya164** [comment]
@bestander Weird. I'm unable to find the cause from the logs. Still trying to figure out why it's failing.

**Comment by bestander** [comment]
It passes on Travis which is Node 4 and npm 2.
Maybe it is npm 3 thing?

**Comment by satya164** [comment]
@bestander Weird, why would NPM version affect moving a gradle file!!!

**Comment by bestander** [comment]
That is completely mad, that is true.
Maybe bloody packager decided to read gradle? :)

**Comment by satya164** [comment]
@bestander Maybe because I didn't merge master to my branch?

**Comment by bestander** [comment]
Try rebasing on master, @satya164.
And don't bother squashing commits, having many would help us track when changes introduce CI failures.

```
     // get entry file complete path (`entryFile` is relative to roots)
      let entryFilePath;
      if (response.dependencies.length > 1) { // skip HMR requests
        const numModuleSystemDependencies =
          this._resolver.getModuleSystemDependencies({dev, unbundle}).length;

        entryFilePath = response.dependencies[
          (response.numPrependedDependencies || 0) +
          numModuleSystemDependencies
        ].path;
      }
```

**Comment by satya164** [comment]
@bestander Rebased. Fingers crossed!

**Comment by bestander** [comment]
Ok, tests now pass

**Comment by bestander** [comment]
@facebook-github-bot shipit

**Comment by foghina** [comment]
Thanks for picking this up @bestander!
